File: src/insertionSort.ts
type InsertType = (a: number[], b: number, c: number) => number;
type Insertion = (a: number[]) => void;
const insert: InsertType = (arr, rightIndex, val) => {
  let current = rightIndex;
  while (current >= 0 && arr[current] > val) {
    arr[current + 1] = arr[current];
    current--;
  }
  return (arr[current + 1] = val);
};

const insertionSort: Insertion = (arr) => {
  for (let i = 0; i < arr.length; i++) {
    insert(arr, i - 1, arr[i]);
  }
};

const arr = [3, 5, -4, 7, 2, 11, 9, 13];
insertionSort(arr);
console.log(arr);

File: src/python/insertion_sort.py
def insert(arr,last_sorted_idx,insertable_val):
    current_pos = last_sorted_idx
    while current_pos >= 0 and arr[current_pos] > insertable_val:
        arr[current_pos + 1] = arr[current_pos]
        current_pos -= 1

    arr[current_pos + 1] = insertable_val

def insertion(arr):
    for i in range(1,len(arr)):
        last_sorted_idx = i - 1
        insertable_val = arr[i]
        insert(arr,last_sorted_idx,insertable_val)

arr = [5,-1,1,9,2]
insertion(arr)
print(arr)
